server/src/test/java/org/kaazing/gateway/server/context/resolve/DefaultClusterContextTest.java

```diff
@@ -234,6 +245,143 @@ public void shouldNotListenOnMulticastInterfaceImplicitly() throws Exception {
     }
 
     @Test
+    public void shouldSendAndReceiveMessagesOnTopics() throws Exception {
+        try {
+            MemberId member1 = new MemberId("tcp", "127.0.0.1", 46943);
+            MemberId member2 = new MemberId("tcp", "127.0.0.1", 46942);
+
+            List accepts = Collections.singletonList(member1);
+            List connects = Collections.singletonList(member2);
+            final String clusterName = this.getClass().getName() + "-cluster1";
+            clusterContext1 = new DefaultClusterContext(clusterName,
+                accepts,
+                connects,
+                null);
+
+            clusterContext2 = new DefaultClusterContext(clusterName,
+                connects,
+                accepts,
+                null);
+
+            clusterContext1.addMembershipEventListener(memberTracker1);
+            clusterContext1.start();
+
+            clusterContext2.addMembershipEventListener(memberTracker2);
+            clusterContext2.start();
+
+            CountDownLatch listenerCalledDifferentMembers = new CountDownLatch(1);
+            ITopic<String> topicMember1 = clusterContext1.getTopic("topic");
+            topicMember1.addMessageListener(m -> {
+                listenerCalledDifferentMembers.countDown();
+            });
+
+            ITopic<String> topicMember2 = clusterContext2.getTopic("topic");
+            topicMember2.publish("test");
+            listenerCalledDifferentMembers.await();
+
+
+
+            ITopic<String> topicMessageListenerMember1 = clusterContext1.getTopic("topic_message_listener_null_pointer");
+            CountDownLatch listenersCalledNullPointer = new CountDownLatch(3);
+            topicMessageListenerMember1.addMessageListener(message -> listenersCalledNullPointer.countDown());
+            topicMessageListenerMember1.addMessageListener(message -> {listenersCalledNullPointer.countDown(); throw new NullPointerException();});
+            topicMessageListenerMember1.addMessageListener(message -> listenersCalledNullPointer.countDown());
+            ITopic<String> topicMessageListenerMember2 = clusterContext2.getTopic("topic_message_listener_null_pointer");
+            topicMessageListenerMember2.publish("msg1");
+            listenersCalledNullPointer.await();
+            assertEquals(1, topicMessageListenerMember2.getLocalTopicStats().getPublishOperationCount());
+            assertEquals(3, topicMessageListenerMember1.getLocalTopicStats().getReceiveOperationCount());
+
+
+            ITopic<String> topicSameListenerMember1 = clusterContext1.getTopic("topic_same_listener");
```

Comment:
This case does not matter to us, we will never do this. Generally I would expect that the tests in this method would be covered by separate test methods in DefaultClusterTopicTest and StandaloneClusterTopicTest, so maybe they are not needed here?